### PR TITLE
Add rustc command to build system

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -118,10 +118,10 @@ error: failed to copy bitcode to object file: No such file or directory (os erro
 
 ### Rustc
 
-> You should prefer using the Cargo method.
+If you want to run `rustc` directly, you can do so with:
 
 ```bash
-$ LIBRARY_PATH="[gcc-path value]" LD_LIBRARY_PATH="[gcc-path value]" rustc +$(cat $CG_GCCJIT_DIR/rust-toolchain | grep 'channel' | cut -d '=' -f 2 | sed 's/"//g' | sed 's/ //g') -Cpanic=abort -Zcodegen-backend=$CG_GCCJIT_DIR/target/release/librustc_codegen_gcc.so --sysroot $CG_GCCJIT_DIR/build_sysroot/sysroot my_crate.rs
```
Comment (antoyo) [inline Readme.md:124]: I think it would be good to keep this doc, but you can hide it in the doc directory.

File: build_system/src/rust_tools.rs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    get_toolchain, run_command_with_output_and_env_no_err, rustc_toolchain_version_info,
+    rustc_version_info,
+};
+
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+fn args(command: &str) -> Result<Option<Vec<String>>, String> {
+    // We skip the binary and the "cargo" option.
```
Comment (antoyo) [inline build_system/src/rust_tools.rs]:
```suggestion
    // We skip the binary and the "cargo"/"rustc" option.
```

File: build_system/src/rust_tools.rs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    toolchain: String,
+    config: ConfigInfo,
+}
+
+impl RustcTools {
+    fn new(command: &str) -> Result<Option<Self>, String> {
+        let args = match args(command)? {
+            Some(a) => a,
+            None => return Ok(None),
+        };
```
Comment (antoyo) [inline build_system/src/rust_tools.rs]: `let … else` ?

File: Readme.md (hunk 1 above)
Comment (GuillaumeGomez) [reply Readme.md:124]: Hum ok.

File: build_system/src/rust_tools.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply build_system/src/rust_tools.rs]: Haha, yes sir!

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by antoyo** [comment]
Thanks for your contribution!

**Review by antoyo — APPROVED** [review]

**Pull request merged**